flow/connectors/connelasticsearch/qrep.go

```diff
@@ -0,0 +1,175 @@
+package connelasticsearch
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"slices"
+	"sync"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v8/esutil"
+	"github.com/google/uuid"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/model"
+	"github.com/PeerDB-io/peer-flow/model/qvalue"
+	"github.com/PeerDB-io/peer-flow/shared"
+)
+
+func (esc *ElasticsearchConnector) SetupQRepMetadataTables(ctx context.Context,
+	config *protos.QRepConfig,
+) error {
+	return nil
+}
+
+func upsertKeyColsHash(qRecord []qvalue.QValue, upsertColIndices []int) string {
+	upsertColsMerged := make([][]byte, 0, len(upsertColIndices))
+
+	for _, upsertColIndex := range upsertColIndices {
+		upsertColsMerged = append(upsertColsMerged, []byte(fmt.Sprint(qRecord[upsertColIndex].Value())))
+	}
+	hashBytes := sha256.Sum256(slices.Concat(upsertColsMerged...))
```

Comment:
https://pkg.go.dev/crypto/sha256#New returns `hash.Hash` which implements io.Writer so loop could use `hasher.Write(...)` instead of appending & at end `return hex.EncodeToString(hasher.Sum(nil))`